src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -678,5 +677,10 @@ private DateTimeOffset AddRandomDelay(DateTimeOffset dt, TimeSpan maxDelay)
             long randomTicks = (long)(maxDelay.Ticks * RandomGenerator.NextDouble());
             return dt.AddTicks(randomTicks);
         }
+
+        private bool IsAuthenticationError(RequestFailedException ex)

```

Comment:
What about a pattern that allows the catch blocks to be identical.
```
bool IsAuthenticationError(Exception e)
{
  if (e is RequestFailedException rfe)
  {
    return ex.Status == (int)HttpStatusCode.Unauthorized || ex.Status == (int)HttpStatusCode.Forbidden;
  }

  if (e is AggregateException ae)
  {
    return ae.InnerExceptions?.Any(inner => IsAuthenticationError(inner));
  }

  return false;
}
```

Reply:
Updated, thanks!